Build fails with modern compilers

Cloning the latest code from git master, and building with the following *FLAGS: `-Werror=implicit-function-declaration -Werror=implicit-int -Werror=int-conversion -Werror=incompatible-pointer-types`

These warnings are becoming default errors in GCC 14 and clang 16. They indicate code that was (usually) valid C89, but removed from the standard in C99 because they were so dangerous. Compilers are finally starting to enforce this, so code that uses it will not compile anymore after upgrading to the newest compilers, even if no -Werror is passed.

dfshow failed to build:

```
gcc -DHAVE_CONFIG_H -I.  -DLOCALEDIR=\"/usr/local/share/locale\"  -DSYSCONFIG=\"/usr/local/etc\" -DDATADIR=\"/usr/local/share/dfshow\" -D_XOPEN_SOURCE_EXTENDED -fno-common -DAPPLICATION_SHOW -DAPPLICATION_SF -march=native -fstack-protector-all -O2 -pipe -fdiagnostics-color=always -frecord-gcc-switches -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=3 -fstack-clash-protection -flto=4 -Werror=odr -Werror=lto-type-mismatch -Werror=strict-aliasing  -Werror=implicit-function-declaration -Werror=implicit-int -Werror=int-conversion -Werror=incompatible-pointer-types -MT src/bin_show-input.o -MD -MP -MF src/.deps/bin_show-input.Tpo -c -o src/bin_show-input.o `test -f 'src/input.c' || echo './'`src/input.c
src/input.c: In function ‘wReadLine’:
src/input.c:73:5: error: implicit declaration of function ‘get_wch’; did you mean ‘getwc’? [-Werror=implicit-function-declaration]
   73 |     get_wch(&c);
      |     ^~~~~~~
      |     getwc
cc1: some warnings being treated as errors
make[2]: *** [Makefile:920: src/bin_show-input.o] Error 1
gcc -DHAVE_CONFIG_H -I.  -DLOCALEDIR=\"/usr/local/share/locale\"  -DSYSCONFIG=\"/usr/local/etc\" -DDATADIR=\"/usr/local/share/dfshow\" -D_XOPEN_SOURCE_EXTENDED -fno-common -DAPPLICATION_SF -march=native -fstack-protector-all -O2 -pipe -fdiagnostics-color=always -frecord-gcc-switches -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=3 -fstack-clash-protection -flto=4 -Werror=odr -Werror=lto-type-mismatch -Werror=strict-aliasing  -Werror=implicit-function-declaration -Werror=implicit-int -Werror=int-conversion -Werror=incompatible-pointer-types -MT src/bin_sf-input.o -MD -MP -MF src/.deps/bin_sf-input.Tpo -c -o src/bin_sf-input.o `test -f 'src/input.c' || echo './'`src/input.c
src/input.c: In function ‘wReadLine’:
src/input.c:73:5: error: implicit declaration of function ‘get_wch’; did you mean ‘getwc’? [-Werror=implicit-function-declaration]
   73 |     get_wch(&c);
      |     ^~~~~~~
      |     getwc
cc1: some warnings being treated as errors
make[2]: *** [Makefile:752: src/bin_sf-input.o] Error 1
```

--- a/Makefile.am
+++ b/Makefile.am
@@ -33,10 +33,12 @@ endif
 bin_PROGRAMS = bin/show bin/sf
 bin_show_SOURCES = src/show.c src/showfunctions.c src/sffunctions.c src/showmenus.c src/sfmenus.c src/colors.c src/common.c src/menu.c src/input.c src/display.c src/settings.c src/i18n.c
 bin_show_LDADD = $(LDADD)
-bin_show_CFLAGS = $(EXTRA_CFLAGS) $(AM_CFLAGS) -DAPPLICATION_SHOW -DAPPLICATION_SF
+bin_show_CFLAGS = $(NCURSES_CFLAGS) $(EXTRA_CFLAGS) $(AM_CFLAGS) -DAPPLICATION_SHOW -DAPPLICATION_SF
+bin_show_LDFLAGS = $(NCURSES_LIBS) $(AM_LDFLAGS)
 bin_sf_SOURCES = src/sf.c src/sffunctions.c src/sfmenus.c src/colors.c src/common.c src/menu.c src/input.c src/display.c src/settings.c src/i18n.c
+bin_sf_LDFLAGS = $(NCURSES_LIBS) $(AM_LDFLAGS)
 bin_sf_LDADD = $(LDADD)
-bin_sf_CFLAGS = $(EXTRA_CFLAGS) $(AM_CFLAGS) -DAPPLICATION_SF
+bin_sf_CFLAGS = $(NCURSES_CFLAGS) $(EXTRA_CFLAGS) $(AM_CFLAGS) -DAPPLICATION_SF
 dfshowconf_DATA = conf/dfshow.conf
 dfshowdata_DATA = themes/*
 

--- a/configure.ac
+++ b/configure.ac
@@ -41,30 +41,14 @@ AC_CHECK_DECL([ACL_TYPE_EXTENDED],
     #include <sys/acl.h>])
 # Solaris and derivates ACLs
 AC_CHECK_LIB([sec], [acl_get])
-# Check for ncurses with wide character support
-AC_CHECK_LIB([ncursesw], [initscr], [],
-            [AC_MSG_ERROR([ncurses with wide character support not found.])])
 AC_CHECK_TYPES([aclent_t], [], [], [[#include <sys/acl.h>]])
 AC_CHECK_TYPES([ace_t], [], [], [[#include <sys/acl.h>]])
 AC_CHECK_FUNCS(acl_get facl_get acl_set facl_set)
 
-AC_CHECK_TOOL([UNAME], [uname], [false])
-case `$UNAME -s` in
-    Darwin*)
-        # macOS to use Homebrew version of ncurses which is v6
-        NCURSES_PREFIX=`brew --prefix ncurses`
-
-        CPPFLAGS="-I$NCURSES_PREFIX/include $CPPFLAGS"
-        LDFLAGS="-L$NCURSES_PREFIX/lib $LDFLAGS"
-        ;;
-
-    *)
-        ;;
-esac
+PKG_CHECK_MODULES([NCURSES], [ncursesw])
 
 AC_CHECK_MEMBERS([struct stat.st_author])
 AC_CHECK_HEADERS([stdio.h limits.h signal.h ctype.h wctype.h getopt.h sys/types.h sys/stat.h dirent.h fcntl.h pwd.h string.h stdlib.h unistd.h time.h sys/statvfs.h libgen.h errno.h wchar.h hurd.h math.h sys/sysmacros.h regex.h utime.h sys/xattr.h acl/libacl.h stdint.h])
-AC_CHECK_HEADERS(ncurses.h, , AC_MSG_ERROR(ncurses header (ncurses.h) not found. You may need to install an ncurses development package.))
 AC_CHECK_HEADERS(libconfig.h, , AC_MSG_ERROR(libconfig header (libconfig.h) not found. You may need to install a libconfig development package.))
 AC_CHECK_HEADERS(sys/acl.h, , AC_MSG_ERROR(libacl header (sys/acl.h) not found. You may need to install a libacl development package.))
 

